iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/api/ResourceDeclaration.java

```diff
@@ -29,5 +29,7 @@ public interface ResourceDeclaration extends Statement, HasProperties {
 
   StringLiteral type();
 
+  String fullType();
```

Comment:
What does full type mean?

Reply:
The purpose of `fullType` is to retrieve the complete string of a resource independently of it being defined at root level or within a parent resource.

As per [ARM documentation](https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/child-resource-name-type):
> You don't provide {parent type prefix} because it's assumed from the parent resource type.

We have a difference in how the `type` string is provided depending if a child resource is injected directly in the parent resource or if it is created aside.

See this example:
```
"resources": [
  {
    "type": "Microsoft.Network/virtualNetworks",
    "apiVersion": "2018-10-01",
    "name": "VNet1",
    "location": "[parameters('location')]",
    "properties": {
      "addressSpace": {
        "addressPrefixes": [
          "10.0.0.0/16"
        ]
      }
    },
    "resources": [
      {
        "type": "subnets",
        "apiVersion": "2018-10-01",
        "name": "Subnet1",
        "dependsOn": [
          "VNet1"
        ],
        "properties": {
          "addressPrefix": "10.0.0.0/24"
        }
      }
    ]
  }
]
```
Notice here that the `type` of the injected child resource is only `subnets` and does not contain the parent `type`.

See this other example:
```
"resources": [
  {
    "type": "Microsoft.Network/virtualNetworks",
    "apiVersion": "2018-10-01",
    "name": "VNet1",
    "location": "[parameters('location')]",
    "properties": {
      "addressSpace": {
        "addressPrefixes": [
          "10.0.0.0/16"
        ]
      }
    }
  },
  {
    "type": "Microsoft.Network/virtualNetworks/subnets",
    "apiVersion": "2018-10-01",
    "name": "VNet1/Subnet1",
    "dependsOn": [
      "VNet1"
    ],
    "properties": {
      "addressPrefix": "10.0.0.0/24"
    }
  }
]
```
Here for the `type` of the child resource, we have the full "path" of the type: `Microsoft.Network/virtualNetworks/subnets`

This two examples describe the same infrastructure and resource declaration, written in two different way.

The goal of this `fullType()` method was to provide in both case the string value `Microsoft.Network/virtualNetworks/subnets`, so that our checks can rely on this method to detect eligible resource to rules.

Let me know if this is not clear.
Probably `fullType()` could be renamed to something more meaningful.

Reply:
I ended up removing the bidirectional relationship, now that I see the different approach (from top to bottom), it doesn't make sense anymore to have it.